tornado/platform/common.py

```diff
@@ -3,10 +3,26 @@
 
 import errno
 import socket
+import time
 
 from tornado.platform import interface
 
 
+def try_close(f):
+    # Avoid issue #875 (race condition when using the file in another
+    # thread).
+    for i in range(10):
+        try:
+            f.close()
+        except IOError:
+            # Yield to another thread
+            time.sleep(1e-3)
```

Comment:
Unless many non-Python threads are competing for the CPU, yes. We cannot guarantee this approach will always work, but it reasonably should.